### DISCUSSION TITLE
Welcome to ripple Discussions!

### DISCUSSION THREAD

**Discussion by spleenware** [spleenware/ripple#64] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by cracky22** [comment]
Hello Scott. Thank you for the great work. A good friend and I (both 16 years old) really like your project. We would like to assemble such a chat device, but we are still a little unsure about how it works because the source code is unfortunately nowhere to be found. In the second model (with ultra energy-saving mode) can everything be set via the display? I would also be interested to know whether the device also displays long messages. Can I theoretically also install a mute switch between the buzzer?

**Reply by spleenware** [reply]
This article is your best bet:  https://www.hackster.io/scottpowell69/an-even-better-lora-qwerty-messenger-bf1cd2
Yes, you can do all of the config via on-screen menus.
Yes, messages that are too long are automatically scrolled horizontally (in the conversation screen).
You could put a physical switch in between either of the buzzer wires to make a physical mute.

**Reply by cracky22** [reply]
Hello Scott, I have assembled my device so far, however, without a beeper and without a real-time clock. Unfortunately, my keyboard (which you linked on aliexpress) is not recognized. I soldered them for the second time. at every start it flashes briefly (keyboard lighting) and the ripple logo appears briefly. after that it says KEY MISSING but when I press enter nothing happens.

**Reply by spleenware** [reply]
Which development board are you using? The Heltec V2 like in the article? There is a newer V3, and just wondering if you might be using that(?) It has a USB-C connector instead of the older micro.

**Reply by cracky22** [reply]
no currently im using the v2

**Reply by spleenware** [reply]
All I can think of is to just double-check your wiring. (SDA to pin 4, SCL to pin 15, etc).
Actually, try wiring the keyboard's 3V3 (ie. power) to the Heltec 3.3v, not the Vext pin. (is on opposite side to Vext).

**Reply by cracky22** [reply]
![20230710_154216](https://github.com/spleenware/ripple/assets/104385850/ca092768-e8c6-4441-9adc-c6ee052e87c3)
 It doesnt work...

**Reply by spleenware** [reply]
Wiring looks correct, but the controller board looks very different to the one I have used. :-( I suspect it's some new model, and has some different I2C protocol now. If you can get info on it, and whether there is an Arduino interfacing library for it, I could try to make a custom Ripple firmware.

**Reply by cracky22** [reply]
Unfortunately I do not know it. this is the board you linked via aliexpress-link: https://www.aliexpress.com/item/1005004730160414.html

**Reply by cracky22** [reply]
maybe this will help you
![Screenshot_20230711_070132_Chrome](https://github.com/spleenware/ripple/assets/104385850/e830f26e-87e8-4ed2-b622-c9c901f0f243)

**Reply by spleenware** [reply]
The best I can suggest is that you'll have to get lower-level with this particular hardware, and try to find what works. If you have the Arduino IDE, try running the 'Polling.ino' sketch in the examples folder of this repo:
    https://github.com/arturo182/arduino_bbq10kbd/tree/master

If this doesn't work, then I don't know what will, unfortunately.

**Reply by cracky22** [reply]
I don't quite understand.. I have now downloaded this file: https://github.com/arturo182/arduino_bbq10kbd/blob/master/examples/Polling/Polling.ino and opened it in the arduino ide. should I now connect and select the heltec board or how can I install the heltec v2 in the board manager?

**Comment by PirateNinja777** [comment]
Hello Scott,

I was wondering if you have any plans to work on firmware for the T-watch? I am setting up a Lora mesh network for my kids when we go camping. 

Thank you for all your hard work!
